iis/mymodule.cpp

```diff
@@ -816,6 +826,9 @@ CMyHttpModule::OnBeginRequest(IHttpContext* httpContext, IHttpEventProvider* pro
         hr = config->GlobalWideCharToMultiByte(config->GetPath(), wcslen(config->GetPath()), &path, &pathlen);
         if (FAILED(hr))
         {
+            config->configLoadingFailed = true;
```

Comment:
The pattern of four line: 
```
            config->configLoadingFailed = true;
            httpContext->GetResponse()->SetStatus(500, "WAF configuration error");
            httpContext->SetRequestHandled();
            return RQ_NOTIFICATION_FINISH_REQUEST;
```

occurs three times in this code. Suggest wrapping it into a helper lambda, something like:

```
auto reportConfigurationError = [config, httpContext] {
            config->configLoadingFailed = true;
            httpContext->GetResponse()->SetStatus(500, "WAF configuration error");
            httpContext->SetRequestHandled();
            return RQ_NOTIFICATION_FINISH_REQUEST;
}

...

if (FAILED(hr)) {
    return reportConfigurationError();
}
```.

This will avoid code repetition and make it easier to update all the places, e.g., to change the error message.

Reply:
very nice! Had never used lambda's in C++ before!